astropy/stats/histogram.py

```diff
@@ -10,7 +10,90 @@
 from . import bayesian_blocks
 
 __all__ = ['histogram', 'scott_bin_width', 'freedman_bin_width',
-           'knuth_bin_width']
+           'knuth_bin_width', 'calculate_bin_edges']
+
+
+def calculate_bin_edges(a, bins=10, range=None, weights=None):
+    """
+    Calculate histogram bin edges like `numpy.histogram_bin_edges`.
+
+    Parameters
+    ----------
+
+    a : array_like
+        Input data. The bin edges are calculated over the flattened array.
+
+    bins : int or list or str (optional)
+        If ``bins`` is an int, it is the number of bins. If it is a list
+        it is taken to be the bin edges. If it is a string, it must be one
+        of  'blocks', 'knuth', 'scott' or 'freedman'. See
+        `~astropy.stats.histogram` for a description of each method.
+
+    range : tuple or None (optional)
+        The minimum and maximum range for the histogram.  If not specified,
+        it will be (x.min(), x.max()). However, if bins is a list it is
+        returned unmodified regardless of the range argument.
+
+    weights : array_like, optional
```

Comment:
Sorry for a random comment from the sidelines, but why include `weights` if it does not influence the calculation?

Reply:
I think it was because of this statement in [`np.histogram_bin_edges`](https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.histogram_bin_edges.html), which takes a `weights` argument:

> weights : array_like, optional
>> An array of weights, of the same shape as a. Each value in a only contributes its associated weight towards the bin count (instead of 1). This is currently not used by any of the bin estimators, but may be in the future.

Right now it does no harm, but if, in the future, there is some sort of way of finding histogram edges in numpy that uses weights we won't need to make any changes.

Reply:
OK, if we just mimic `np.histogram_bin_edges`, then that's totally fine. Maybe add the same sentence, that it might be used in future.